### PR TITLE
chore(manifests): Migrate manifests from downstream repo

### PR DESCRIPTION
**Pull request by dgutride**
## Description

Migrate  RHOAI manifests from the downstream repo into the manifests folder here.  Some items will need to become overlays and also require some additional handling by the installer which will be addressed as we move forward.

## How Has This Been Tested?

Run kustomize build to verify the yaml files.

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Comment by lucferbux** [comment]
@VaishnaviHire @zdtsw this is the draft for the manifest migration, we need to do some tweaks but so far we have part of the changes here

File: manifests/overlays/rhoai/params.env [hunk 1]
```diff
@@ -0,0 +1 @@
+odh-dashboard-image=quay.io/opendatahub/odh-dashboard:main
```
Comment (zdtsw) [inline manifests/overlays/rhoai/params.env:1]: so the file is change from 	`manifests/overlay/rhods/params.env`
to `manifests/overlays/rhoai/params.env`

we will need am update in operator for this

File: manifests/base/kustomization.yaml [hunk 2]
```diff
@@ -5,7 +5,7 @@ commonLabels:
   app.kubernetes.io/part-of: odh-dashboard
 resources:
   - ../apps
-  - ../modelserving
+  - ../crd
```
Comment (zdtsw) [inline manifests/base/kustomization.yaml:8]: do you still deploy OVMS or it is just changed from modelserving folder to crd? 
if it is no OVMS, we will need to cleanup the logic in opreator

Comment (lucferbux) [reply manifests/base/kustomization.yaml:8]: Is there a specific logic for OVMS? We can check that today, we are moving away that folder completely and transering ownership to the modelserving team. But we can get something in place for this PR

File: manifests/overlays/rhoai/params.env (hunk 1 above)
Comment (lucferbux) [reply manifests/overlays/rhoai/params.env:1]: Yess! That's one of the things I was gonna ping you about, I'm doing the list of changes for the platform to be aware of!

File: manifests/overlays/rhoai/kustomization.yaml [hunk 3]
```diff
@@ -0,0 +1,86 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+commonLabels:
+  app: rhods-dashboard
+  app.kubernetes.io/part-of: rhods-dashboard
+bases:
```
Comment (lucferbux) [inline manifests/overlays/rhoai/kustomization.yaml:6]: btw @zdtsw @VaishnaviHire we kept bases here cause it was giving issues in v1 operator, can we change it to `resources already?`

Comment (zdtsw) [reply manifests/overlays/rhoai/kustomization.yaml:6]: we have not updated kustomize to v5, but i think `resources` should work, can try that

**Comment by lucferbux** [comment]
## Overview

- Moved rhods overlay to manifests/overlays and renamed it to rhea
- Moved console link to manifests and add an overlay
- The only change for upstream would be deploying the consoling and injecting the url such as we do in rhoai
- We are deprecating odhdashboardconfig we shouldn’t deploy it by default, the dashboard might create one if missing
- In console link, we’ve changed spec.href from <rhods-dashboard-url> to <dashboard-url>


## Open questions

- What happens with odh-enabled-applications-config.configmap.yaml ?? this is the other file im not sure what to do
- Can we change the base attribute to resources now that we have v2??
- Do we have an icon for odh console link??


## What should the operator deploy

### ODH

- manifest/base ✅
- manifest/modelserving ✅
- manifest/consolelink ✅
- manifest/apps ✅


#### RHOAI

- manifest/overlays/rhoai ✅
- manifest/overlays/modelserving ✅
- manifest/overlays/consolelink ✅
- For apps
    - add-on: manifests/overlays/apps-addon ✅
    - on-prem: manifests/overlays/apps-onprem ✅

**Comment by lucferbux** [comment]
@zdtsw @vaibhavjainwiz here's the proposed changes, let me know what you think or if it's more or less what you had in mind, we are leaning to get everything in overlays so it's fairly easy for us to keep track of everything.

On the other side, we are expecting to transfer the `movelserving` folder (with ovms and more) to the modelserving in the near future, but this is a stopgap that will keep the serving runtimes in a healty state until we do that migration

**Comment by lucferbux** [comment]
> > @VaishnaviHire @zdtsw this is the draft for the manifest migration, we need to do some tweaks but so far we have part of the changes here
> 
> 62 files and we have ODH 2.8 code freeze in 2days :P

wait what? isn't code freeze next week???? o.O

**Comment by dgutride** [comment]
I've merged in your commit @lucferbux  - we might need to revisit this with a squash commit - let me know

File: manifests/consolelink/consolelink.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  href: <dashboard-url>
+  location: ApplicationMenu
+  text: Open Data Hub
```
Comment (zdtsw) [inline manifests/consolelink/consolelink.yaml:11]: 2 values "href" and "text":
1. i saw you have the overlay of rhoai, to replace text, which should cover for both ODH and rhoai already
2. i dont know if the plan is to get consolelink for both ODH and downstream assume this is the reason changed from <rhods-dashboard-url> to <dashboard-url> for "href"
then operator need to do some updates for this part. just need to confirm, you have decided to use dashboard-url as the placeholder here

File: manifests/overlays/modelserving/params.yaml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+# varReference:
```
Comment (zdtsw) [inline manifests/overlays/modelserving/params.yaml:1]: still wanna keep this file?

**Review by zdtsw — COMMENTED** [review]
in

**Comment by zdtsw** [comment]
i think, to narrow down the scope of this PR if the goal is to deprecate odh-manifests then we should not do changes for consolelink's content.
this relates to my comments + your open question to "Do we have an icon for odh console link??"

if all test is fine with this PR, we can have another ticket on "introduce consolelink to ODH" (which i dont see anyone would object it, but as you said we might need the new encode icon from someone else)

**Comment by zdtsw** [comment]
> ## Overview
> * Moved rhods overlay to manifests/overlays and renamed it to rhea
> * Moved console link to manifests and add an overlay
> * The only change for upstream would be deploying the consoling and injecting the url such as we do in rhoai
> * We are deprecating odhdashboardconfig we shouldn’t deploy it by default, the dashboard might create one if missing
> * In console link, we’ve changed spec.href from  to
> 
> ## Open questions
> * What happens with odh-enabled-applications-config.configmap.yaml ?? this is the other file im not sure what to do
> * Can we change the base attribute to resources now that we have v2??
> * Do we have an icon for odh console link??
> 
> ## What should the operator deploy
> ### ODH
> * manifest/base ✅
> * manifest/modelserving ✅
> * manifest/consolelink ✅
> * manifest/apps ✅
> 
> ### RHOAI
> * manifest/overlays/rhoai ✅
> * manifest/overlays/modelserving ✅
> * manifest/overlays/consolelink ✅
> * For apps
>   
>   * add-on: manifests/overlays/apps-addon ✅
>   * on-prem: manifests/overlays/apps-onprem ✅

do we still deploy "crd"?

**Comment by zdtsw** [comment]
one more thing: we do still deploy anaconda-ce-periodic-validator in downstream? 
it is part of the config living in downstream rhoai operator https://github.com/zdtsw-forking/rhods-operator/blob/main/config/partners/anaconda/base/anaconda-ce-validator-cron.yaml, if you want to have it in your overlays/rhoai we can cleanup it from operator

**Comment by lucferbux** [comment]
Oh and @dgutride we might wanna include this info somewhere in the dashboard's docs: https://github.com/red-hat-data-services/odh-deployer/blob/main/CONTRIBUTING.md#isv-contribution

File: manifests/overlays/modelserving/params.yaml (hunk 5 above)
Comment (lucferbux) [reply manifests/overlays/modelserving/params.yaml:1]: nope, removing it later

**Comment by lucferbux** [comment]
> one more thing: we do still deploy anaconda-ce-periodic-validator in downstream? it is part of the config living in downstream rhoai operator https://github.com/zdtsw-forking/rhods-operator/blob/main/config/partners/anaconda/base/anaconda-ce-validator-cron.yaml, if you want to have it in your overlays/rhoai we can cleanup it from operator

Great chat!

**Comment by lucferbux** [comment]
Closing this PR in favor of https://github.com/opendatahub-io/odh-dashboard/pull/2519

**Pull request closed without merging**